instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpCommonAttributesExtractor.java

```diff
@@ -35,19 +38,29 @@ abstract class HttpCommonAttributesExtractor<
   final GETTER getter;
   private final List<String> capturedRequestHeaders;
   private final List<String> capturedResponseHeaders;
+  private final Set<String> knownMethods;
 
   HttpCommonAttributesExtractor(
-      GETTER getter, List<String> capturedRequestHeaders, List<String> capturedResponseHeaders) {
+      GETTER getter,
+      List<String> capturedRequestHeaders,
+      List<String> capturedResponseHeaders,
+      Set<String> knownMethods) {
     this.getter = getter;
     this.capturedRequestHeaders = lowercase(capturedRequestHeaders);
     this.capturedResponseHeaders = lowercase(capturedResponseHeaders);
+    this.knownMethods = new HashSet<>(knownMethods);
   }
 
   @Override
   public void onStart(AttributesBuilder attributes, Context parentContext, REQUEST request) {
     String method = getter.getHttpRequestMethod(request);
     if (SemconvStability.emitStableHttpSemconv()) {
-      internalSet(attributes, HttpAttributes.HTTP_REQUEST_METHOD, method);
+      if (knownMethods.contains(method)) {
+        internalSet(attributes, HttpAttributes.HTTP_REQUEST_METHOD, method);
+      } else {
+        internalSet(attributes, HttpAttributes.HTTP_REQUEST_METHOD, _OTHER);
```

Comment:
static import the names to improve readability?

Reply:
The `HttpAttributes` values? I'm in favor of that actually -- but if we're doing that, we should implement it for all attribute keys across the whole package. I think I'd prefer to make it a separate PR, if you don't mind